### DISCUSSION TITLE
bug in Secure Fields cardNumber in validityChange event

### DISCUSSION THREAD

**Discussion by ihojmanb** [mercadopago/sdk-js#109] [Q&A] [answered]
I'm having trouble with card validation, and I think it's possibly a bug, but correct me if I'm wrong.

I'm using the following code to check the validity of every secure field:
```js
      secureFieldElement.on("validityChange", (data) => {
        const { field, errorMessages } = data;
        console.log(`${field}:errorMessages:`, errorMessages);
        if (errorMessages[0]) {
          console.log(`there is an error with ${field}`);
          // ...
          // do something with the error
        } else {
          console.log("no errors presented");
         // do something else
        }
      });
```
(I found that code somewhere in the docs, not sure where)

So when I use the test card number available in the documentation, say `5416 7526 0258 2580`, and type the first 8 digits (what is known as the *bin*) I get no `errorMessages`. This will be understood by my program as if the typed digits are a valid card number, hence bypassing the form validation, enabling my user to submit an incorrect card information.

If I delete the last digit -- now having only 7 digits in the input field -- I receive the following error message object:
```
{cause: 'invalid_length', message: "cardNumber should be of length '16'."}
```
If I retype the deleted message, this error object repeats.

So my question is, why is this not the expected behavior from the beginning?

Here is a sequence of the events described:
1) first 8 digits
![image](https://user-images.githubusercontent.com/30912229/203367348-124a48c1-49be-41e9-b22d-fcac3a057803.png)
2) delete last digit
![image](https://user-images.githubusercontent.com/30912229/203368023-8deaa3c7-88de-4231-b59b-fc4373ed1c79.png)
3) retype last digit
![image](https://user-images.githubusercontent.com/30912229/203367623-58a85d2e-7205-4856-9fd5-16449c0eb627.png)

**Comment by ihojmanb** [comment]
To endorse this, if we type `8888 8888`, which is an invalid *bin*, the validatityChange event does not detect any errors

**Reply by ihojmanb** [reply]
the same goes with `8888 8888 88` which is neither a valid card number, but the validation does not seem to apply either

**Comment by icaldana** [comment, marked as answer]
Hello @ihojmanb! 

We apologize for the delay and for the inconvenience it may have caused to your workflow. Let's clarify some points:

- Not every card has 16 digits, the market range is from 8 to 19 characters, depending on the card network.
- With the first 8 characters, aka the bin, we can identify the card network and then know how many digits that card must have. So until 7 digits, we only know it should have between 8 and 19 numbers.

I was performing some testes, and I've noticed the behavior I've described isn't always happening, therefore, an internal bug ticket as created to polish this behavior. I would like to thank you for reporting this and I promise I will handle it with care.

Thank you.

**Reply by ihojmanb** [reply]
thanks for the support @icaldana, appreciate it!
